### PR TITLE
Refactor Module

### PR DESCRIPTION
**Pull request by HoshinoTented**
This PR redesigned the module system:
* ~~`import data::nat` can be referred by `nat` rather than `data::nat`~~
* ambiguous on exported symbol is not allowed, therefore we don't need to solve ambiguity when importing ~~, so `UseHide.Name` should store a single string rather than `QualifiedID`~~
* better candidate implementation (a dedicated class rather than a list/map)

### CODE REVIEW

**Comment by ice1000** [comment]
Do you intend to supersede `ModuleExport` with `ModuleExport2`? (and also the other similarly named classes)

**Comment by HoshinoTented** [comment]
> Do you intend to supersede `ModuleExport` with `ModuleExport2`? (and also the other similarly named classes)

Yes, I keep it because I am too lazy to delete it.

**Comment by HoshinoTented** [comment]
The code compiles. However, there are some problems: We have a lot of usage of `import xxx::base`, and they cause name conflicts since the name `base`.

**Comment by HoshinoTented** [comment]
Well, a lot of code is reverted cause:
* We need `ModuleName.Qualified` as the name of module (see last comment)
* I was trying to keep the tree structure of modules, but I found that we can rebuild it from flattened module names, except we lost the layer information (we can not tell whether a module is directly or indirectly importing, but `ResolveInfo` can tell)

**Comment by HoshinoTented** [comment]
However, I found an unnatural place in `CompiledModule::SerExport`, and I fixed it in this PR.

File: base/src/main/java/module-info.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
   requires transitive aya.syntax;
 
   requires static org.jetbrains.annotations;
+  requires jdk.compiler;
```
Comment (ice1000) [inline base/src/main/java/module-info.java:5]: Marked as TODO

**Pull request merged**